### PR TITLE
#68 Implement vendor and admin order management feature

### LINKED ISSUE #68: Vendor and Admin Order Management
## **The description**
Developing order management system for both Vendors and Admins. The  feature will allow vendors to manage their orders and  admins to oversee all orders within the system.

### Acceptance Criteria

- [ ] Vendor order management.
- [ ] Admin order management.
- [ ] State management.
- [ ] API Integration.
- [ ] User interface.
- [ ] Testing.
## Developer Notes

- Define action types and action creators for order operations.
- Implement functions to fetch, create, update and delete orders from the backend.
- Ensure that the components are responsive and work well on different screen sizes.
- Use vitest to write integration tests for the components. 

## Related issues

- issue #69 
- issue #67

### PR DESCRIPTION
**Pull request by aimedivin**
#### What does this PR do?
This PR implements order management functionalities for both vendors and admins. It allows them to view and manage orders placed by buyers and perform actions on these orders where vendor is required to fullfil the request from buyer, while admins  control the interactions between buyers and vendors.

#### Description of Task to be completed?
- implement UI for both vendor and admin order management including viewing all orders and single order.
- Apply styling to order management components using Tailwind CSS.
- Ensure responsive design for different screen sizes.
- Ensure search and pagination work smoothly 
- Implement API calls for updating and fetching order with it's details.
- Write unit tests to ensure the system works correctly.
#### How should this be manually tested?
- Ensure You have backend service running and set VITE_APP_API_URL in your .env file.
- Clone this repository and navigate to the project directory
  ```bash
  git clone -b ft-register https://github.com/atlp-rwanda/knights-ecomm-fe.git
  cd knights-ecomm-fe
  git checkout ft-order-management
  npm i
  npm run dev
  ```
#### What are the relevant pivotal tracker/Trello stories?
#68 

#### Screenshots (if appropriate)
image
image

### CODE REVIEW

**Comment by faid-terence** [comment]
@aimedivin , @GSinseswa721 
What;s the progress on this PR?

**Comment by aimedivin** [comment]
> @aimedivin , @GSinseswa721 What;s the progress on this PR?

@faid-terence  PR is now ready for review

**Review by Ndevu12 — APPROVED** [review]

**Review by maxCastro1 — APPROVED** [review]

File: src/__test__/utils/formartDate&Time.test.ts
Comment (faid-terence) [inline src/__test__/utils/formartDate&Time.test.ts:1]: @aimedivin 
Remove the console.log

**Review by faid-terence — CHANGES_REQUESTED** [review]

File: src/pages/Orders/SingleAdminOrder.tsx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+          completed: orderStatsStates.completed + 1
+        })
+      );
+
+      toast.success('The order has been successfully updated.');
+      setSubmitFormLoading(false);
+      setIsEditable(false);
+    } catch (error) {
+      console.log(error);
+      if ((error as AxiosError).response && [400, 404].includes((error as AxiosError).response!.status)) {
```
Comment (elijahladdie) [inline src/pages/Orders/SingleAdminOrder.tsx:109]: remove the console please

**Review by elijahladdie — APPROVED** [review]

**Review by MC-Knight — APPROVED** [review]

**Review by faid-terence — APPROVED** [review]
Nice Work @GSinseswa721 and @aimedivin 
PR Approved !

**Pull request merged**